### PR TITLE
Support UI customization for iOS SDK [21739]

### PR DESCRIPTION
**Pull request by lqgioan**
## Description
- Added code for supporting UI customization

## Motivation & Context
Ticket: https://app.shortcut.com/credify/story/21739/feature-ios-support-ui-customization-for-ios-sdk

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: Credify/Credify/Network/Requests+Responses.swift [hunk 1]
```diff
@@ -7,6 +7,7 @@
 
 import Alamofire
 import Foundation
+import SwiftUI
```
Comment (ngo275) [inline Credify/Credify/Network/Requests+Responses.swift]: This should be mistake. Please remove this

File: Credify/Credify/Objects/serviceXConfig.swift [hunk 2]
```diff
@@ -1,29 +1,38 @@
...
+//  serviceXConfig.swift
 //  
 //
 //  Created by Shu on 2022/03/12.
 //
 
 import Foundation
 
+
+/// This is serviceX configiration
```
Comment (ngo275) [inline Credify/Credify/Objects/serviceXConfig.swift]: Typo

File: Credify/Credify/Objects/serviceXConfig.swift [hunk 3]
```diff
@@ -1,29 +1,38 @@
...
     let apiKey: String
     let env: CredifyEnvs
     let appName: String
     let theme: serviceXTheme
     let userAgent: String
     
+    
+    /// This is constructor to create serviceX configuration
+    /// - Parameters:
+    ///   - apiKey: Your API that gets from the Dashboard
```
Comment (ngo275) [inline Credify/Credify/Objects/serviceXConfig.swift]: `API key that you can generate on the Dashboard`

File: Credify/Credify/Objects/serviceXConfig.swift [hunk 4]
```diff
@@ -1,29 +1,38 @@
...
+    ///   - env: Environment that you want to test
```
Comment (ngo275) [inline Credify/Credify/Objects/serviceXConfig.swift]: `Environment that you want to use`

File: Credify/Credify/Theme/ThemeColor.swift [hunk 5]
```diff
@@ -7,43 +7,86 @@
 
 import UIKit
 
-public struct ThemeColor {
-    /// Page’s header / Buttons
-    let primaryBrandyStart: UIColor
+
+/// This is for custom the colors
```
Comment (ngo275) [inline Credify/Credify/Theme/ThemeColor.swift]: `/// Theme customization`

File: Credify/Credify/Theme/ThemeColor.swift [hunk 6]
```diff
@@ -7,43 +7,86 @@
...
+public struct ThemeColor : Codable {
+    /// Page’s header bacground
```
Comment (ngo275) [inline Credify/Credify/Theme/ThemeColor.swift]: typo

File: Credify/Credify/Theme/ThemeColor.swift [hunk 7]
```diff
@@ -7,43 +7,86 @@
...
+    public let primaryBrandyStart: String
     
-    /// Page’s header / Buttons
-    let primaryBrandyEnd: UIColor
+    /// Page’s header bacground
```
Comment (ngo275) [inline Credify/Credify/Theme/ThemeColor.swift]: typo

File: Credify/Credify/Theme/ThemeColor.swift [hunk 8]
```diff
@@ -7,43 +7,86 @@
...
     
     /// Primary text
-    let primaryText: UIColor
+    public let primaryText: String
     
     /// Links / Border-color / Emphasized content / Check box / Radio button / Title
-    let secondaryActive: UIColor
+    public let secondaryActive: String
+    
+    // For disable component color
```
Comment (ngo275) [inline Credify/Credify/Theme/ThemeColor.swift]: disabled

File: Credify/Credify/Theme/ThemeColor.swift [hunk 9]
```diff
@@ -7,43 +7,86 @@
...
-                 secondaryActive: UIColor = ThemeColor.default.secondaryActive,
-                 secondaryText: UIColor = ThemeColor.default.secondaryText,
-                 secondaryComponentBackground: UIColor = ThemeColor.default.secondaryComponentBackground,
-                 secondaryBackground: UIColor = ThemeColor.default.secondaryBackground) {
+    public let secondaryBackground: String
+    
+    /// Buttons text color
+    public let primaryButtonTextColor: String
+    
+    /// Buttons bacground
```
Comment (ngo275) [inline Credify/Credify/Theme/ThemeColor.swift]: typo again

File: Credify/Credify/Theme/ThemeFont.swift [hunk 10]
```diff
@@ -8,45 +8,101 @@
...
-                secondaryBoldText: UIFont = ThemeFont.default.secondaryBoldText) {
-        self.primaryPageTitle = primaryPageTitle
-        self.primaryModalTitle = primaryModalTitle
-        self.primarySectionTitle = primarySectionTitle
-        self.primaryButtonTitle = primaryButtonTitle
-        self.secondaryNormalText = secondaryNormalText
-        self.secondarySmallText = secondarySmallText
-        self.secondaryBigText = secondaryBigText
-        self.secondaryBoldText = secondaryBoldText
+/// This is for custom font
```
Comment (ngo275) [inline Credify/Credify/Theme/ThemeFont.swift]: Font customization

File: Credify/Credify/Theme/serviceXTheme.swift [hunk 11]
```diff
@@ -8,48 +8,50 @@
...
+public struct serviceXTheme : Encodable {
     public let color: ThemeColor
-//    public let font: ThemeFont
-//    public let icon: ThemeIcon
-    
+    public let font: ThemeFont
     public let inputFieldRadius: Double
     public let pageHeaderRadius: Double
-    public let modalRadius: Double
+    public let modelRadius: Double
```
Comment (ngo275) [inline Credify/Credify/Theme/serviceXTheme.swift]: `modal`, no?

**Review by ngo275 — CHANGES_REQUESTED** [review]

File: Credify/Credify/Network/Requests+Responses.swift (hunk 1 above)
Comment (lqgioan) [reply Credify/Credify/Network/Requests+Responses.swift]: I removed it

File: Credify/Credify/Objects/serviceXConfig.swift (hunk 2 above)
Comment (lqgioan) [reply Credify/Credify/Objects/serviceXConfig.swift]: I fixed it

File: Credify/Credify/Objects/serviceXConfig.swift (hunk 3 above)
Comment (lqgioan) [reply Credify/Credify/Objects/serviceXConfig.swift]: I updated it

File: Credify/Credify/Objects/serviceXConfig.swift (hunk 4 above)
Comment (lqgioan) [reply Credify/Credify/Objects/serviceXConfig.swift]: I updated it

File: Credify/Credify/Theme/ThemeColor.swift (hunk 5 above)
Comment (lqgioan) [reply Credify/Credify/Theme/ThemeColor.swift]: I updated it

File: Credify/Credify/Theme/ThemeColor.swift (hunk 6 above)
Comment (lqgioan) [reply Credify/Credify/Theme/ThemeColor.swift]: I fixed it

File: Credify/Credify/Theme/ThemeColor.swift (hunk 7 above)
Comment (lqgioan) [reply Credify/Credify/Theme/ThemeColor.swift]: I fixed it

File: Credify/Credify/Theme/ThemeColor.swift (hunk 8 above)
Comment (lqgioan) [reply Credify/Credify/Theme/ThemeColor.swift]: I updated it

File: Credify/Credify/Theme/ThemeColor.swift (hunk 9 above)
Comment (lqgioan) [reply Credify/Credify/Theme/ThemeColor.swift]: I fixed it

File: Credify/Credify/Theme/ThemeFont.swift (hunk 10 above)
Comment (lqgioan) [reply Credify/Credify/Theme/ThemeFont.swift]: I updated it

File: Credify/Credify/Theme/serviceXTheme.swift (hunk 11 above)
Comment (lqgioan) [reply Credify/Credify/Theme/serviceXTheme.swift]: @ngo275 you're right. I updated it on iOS and I need to update on Passport app too. But it's not used in the passport app yet. I will update in the passport app later

**Review by ngo275 — APPROVED** [review]

**Pull request merged**